src/auxiliary/mpi.jl

```diff
@@ -114,16 +111,19 @@ recursive_sum_abs2(u::Number) = abs2(u)
 # https://github.com/SciML/RecursiveArrayTools.jl
 # However, what you have is good enough for us for now, so we don't need this 
 # additional dependency at the moment.
-recursive_sum_abs2(u::AbstractArray) = mapreduce(recursive_sum_abs2, +, u; init=zero(eltype(eltype(u))))
+function recursive_sum_abs2(u::AbstractArray)
+  mapreduce(recursive_sum_abs2, +, u; init = zero(eltype(eltype(u))))
+end
 
 recursive_length(u::Number) = length(u)
 recursive_length(u::AbstractArray{<:Number}) = length(u)
 recursive_length(u::AbstractArray{<:AbstractArray}) = sum(recursive_length, u)
-function recursive_length(u::AbstractArray{<:StaticArrays.StaticArray{S, <:Number}}) where {S}
+function recursive_length(u::AbstractArray{<:StaticArrays.StaticArray{S, <:Number}}) where {
+                                                                                            S
+                                                                                            }
```

Comment:
This is ugly

Reply:
This is an abomination, I agree. In such cases, we probably need to use `#! format: off` (or change the function signature 😈)